src/core/public/keyboard_shortcut/key_parser.ts

```diff
@@ -0,0 +1,459 @@
+   *   if (keyString === parser.normalizeKeyString("Ctrl+S")) {
+   *     // Execute save action
+   *   }
+   * });
+   *
+   */
+  public getEventKeyString(event: KeyboardEvent): string {
+    const modifierChecks = [
+      { check: event.ctrlKey, key: 'ctrl' },
+      { check: event.altKey, key: 'alt' },
+      { check: event.shiftKey, key: 'shift' },
+      { check: event.metaKey, key: 'cmd' },
+    ];
+
+    const modifiers = modifierChecks
+      .filter(({ check }) => check)
+      .map(({ key }) => this.applyPlatformMapping(key));
+
+    const key = this.normalizeKey(event.key.toLowerCase());
+
+    return this.buildKeyString(modifiers, key);
+  }
+
+  private isModifier(key: string): boolean {
+    return KeyStringParser.MODIFIER_KEYS.has(key.toLowerCase());
+  }
+
+  private normalizeModifier(modifier: string): string | null {
+    const normalized = modifier.toLowerCase();
+
+    const alias = KeyStringParser.MODIFIER_ALIASES.get(normalized);
+    if (alias) return alias;
+
+    return KeyStringParser.VALID_MODIFIERS.has(normalized as ModifierKey) ? normalized : null;
+  }
+
+  private applyPlatformMapping(modifier: string): string {
+    const mappings =
+      this.platform === 'mac'
+        ? KeyStringParser.PLATFORM_MODIFIER_MAPPINGS.mac
+        : KeyStringParser.PLATFORM_MODIFIER_MAPPINGS.other;
+
+    const normalized = modifier.toLowerCase();
+    return mappings[normalized as keyof typeof mappings] || normalized;
+  }
+
+  private buildKeyString(modifiers: string[], key: string): string {
+    if (modifiers.length === 0) return key;
+    if (!key) return modifiers.join('+');
+    return `${modifiers.join('+')}+${key}`;
+  }
+
+  private sortModifiers(modifiers: ModifierKey[]): void {
+    modifiers.sort((a, b) => {
+      const aIndex = KeyStringParser.MODIFIER_INDEX_MAP.get(a) ?? -1;
+      const bIndex = KeyStringParser.MODIFIER_INDEX_MAP.get(b) ?? -1;
+      return aIndex - bIndex;
+    });
+  }
+
+  private normalizeKey(key: string): string {
+    if (key === ' ') {
+      return 'space';
+    }
+
+    const trimmed = key.trim().toLowerCase();
+
+    if (!trimmed) {
+      return '';
+    }
+
+    const specialMapping = SPECIAL_KEY_MAPPINGS[trimmed];
+    if (specialMapping) {
+      return specialMapping;
+    }
+
+    if (trimmed.length === 1) {
+      return trimmed;
+    }
+
+    return trimmed;
+  }
+
+  public isValidKeyString(keyString: string): boolean {
+    if (!keyString || typeof keyString !== 'string') {
+      return false;
+    }
+
+    try {
+      const normalized = this.normalizeKeyString(keyString);
+      const parts = normalized.split('+');
+
+      if (parts.length === 0) {
+        return false;
+      }
+
+      const lastPart = parts[parts.length - 1];
+      if (this.isModifier(lastPart)) {
+        return false;
+      }
+
+      for (let i = 0; i < parts.length - 1; i++) {
+        if (!this.isModifier(parts[i])) {
+          return false;
+        }
+      }
```

Comment:
we can also this:
 
`
 return parts.every(!this.isModifier);
`

Reply:
but wait i'm getting confused, why does every part have to be a modifier for it to be a valid key string?